cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterUtils.java

```diff
@@ -177,7 +177,11 @@ public static CruiseControlEndPoint endPoint(HttpServletRequest request) {
         throw new UserRequestException("Unsupported request method: " + request.getMethod() + ".");
     }
 
-    String path = request.getRequestURI().toUpperCase().replace(REQUEST_URI, "");
+    if (apiUrlPrefix.endsWith("/*")) {
+      // Ignore the last character '*'
+      apiUrlPrefix = apiUrlPrefix.substring(0, apiUrlPrefix.length() - 1);
+    }
+    String path = request.getRequestURI().toUpperCase().replace(apiUrlPrefix.toUpperCase(), "");
```

Comment:
Do we have to do this pruning and upper case conversion for `apiUrlPrefix` for every single request that hits a CC endpoint?

`apiUrlPrefix` is already known at configuration time -- i.e. it won't change throughout the lifetime of this CC instance. So can't we initialize this static util class with the well-formatted `REQUEST_URI` (which is constructed by pruning and upper case conversion of `WebServerConfig.WEBSERVER_API_URLPREFIX_CONFIG` once) and use that here as before?

Reply:
It's not a good practice to use a static class to "cache" any value. A static method should be 100% stateless. So I strongly prefer to just stick with the current implementation with the blow change:
```
    if (!apiUrlPrefix.endsWith("/*")) {
      throw new IllegalArgumentException("API URL prefix should end with \"/*\". Got: " + apiUrlPrefix));
    }
```
In this approach, we need to do this pruning and upper case conversion for every single request. It's not a big deal. If we really want to save this "overhead". We will need to think of some way that does not use the `static` method/class. However, that might over-complicate the whole thing. So, I prefer to stick with this approach.

Reply:
Discussed offline and we decided to stick with the current implementation for now. In the future, we can consider using [Dagger 2](https://dagger.dev/)